include/broker/configuration.hh

```diff
@@ -8,14 +8,10 @@ struct broker_options {
   /// If true, peer connections won't use SSL.
   bool disable_ssl = false;
 
-  /// If true, endpoints will forward incoming messages to peers.
-  bool forward = true;
-
-  /// TTL to insert into forwarded messages. Messages will be droppped once
-  /// they have traversed more than this many hops. Note that the 1st
-  /// receiver inserts the TTL (not the sender!). The 1st receiver does
-  /// already count against the TTL.
-  unsigned int ttl = 20;
+  /// If true, endpoints no longer forward incoming subscriptions and other
+  /// routing-related messages to peers. Setting this flag to `true` turns the
+  /// endpoint into a leaf node that never offers forwarding paths to others.
+  bool disable_forwarding = false;
```

Comment:
To me, the sole point of implementing the ALM transport was to enable fully transparent pub/sub in a Zeek cluster. The original design consequently scraped the `forward` flag altogether.

I'm actually unsure whether we should expose this new flag to the user at all. This flag only came in specifically for the use case of having Zeek Agents run behind a gateway.

Reply:
> Also a minor convenience might be to [[deprecated]] the ttl option and just say it's unused/unnecessary now.

Btw, I tried adding this back as deprecated field:

```c++
  [[deprecated("no longer has any effect")]] unsigned int ttl = 0;
```

Doing this makes the build for Broker very noisy, because it emits lots of:

```
broker/include/broker/configuration.hh|27 col 3| warning: 'ttl' is deprecated: no longer has any effect [-Wdeprecated-declarations]
||   broker_options(const broker_options&) = default;
||   ^
broker/include/broker/configuration.hh|58 col 7| note: in defaulted copy constructor for 'broker::broker_options' first required here
|| class configuration : public caf::actor_system_config {
||       ^
broker/build/debug/../../src/broker-gateway.cc|180 col 31| note: in implicit move constructor for 'broker::configuration' first required here
||   if (auto gw = gateway::make(std::move(cfg), internal, external)) {
||                               ^
broker/include/broker/configuration.hh|23 col 5| note: 'ttl' has been explicitly marked deprecated here
||   [[deprecated("no longer has any effect")]] unsigned int ttl = 0;
||     ^
|| 1 warning generated.
```

That's obviously a nonsensical warning, but fixing this would require making all constructors and the assignment operator non-default and then suppress the warning in the `.cc` file. Since this parameter should get passed in via config anyways, I'm not sure if it's worth it. Your call. 🤷🏻‍♂️

Reply:
`ttl` option: ok don't worry about that, I'll simply call out the removal in release notes

`forward` vs `disable_forwarding` options: whether to expose the new flag at all is an independent question that I don't have a strong opinion about and would say it's fine to leave it as is, but the issue I still want to address is that an option named `forward` used to exist and no longer does, so in case someone had used it, we either need to do something to make sure it still exists (e.g. don't change the name of the option if they at least have some shared semantics) or we need to be able to give them a story about how they may want to proceed.  I can simply plan to describe such a story in the release notes:

> The `forward` option no longer exists.  If you previously disabled that setting, the new `disable_forwarding` option can still be used to accomplish similar message (non)routing behavior.  If you previously used that setting (perhaps by default), no further action should be needed: Broker's topic-based pub/sub layer has the same conceptual goals as before, but now implemented with more robust message routing mechanisms.

But still want to know: is that a good description/advice to give or have I misunderstood whether some amount of behavioral-compatibility can be achieved in that way.

Reply:
> But still want to know: is that a good description/advice to give or have I misunderstood whether some amount of behavioral-compatibility can be achieved in that way.

@rsmmr had the idea of merging ALM for 4.0 (as long as test runs all give green light) but turn forwarding off by default for now and then enabling it with 4.1 to give users some "advance warning". In this case, I'd bring back the `forward` flag, so that people have to set this to *on* to opt-in again. With 4.1, we then can either rename the variable to reflect the default behavior or leave as-is. I think renaming the option to `disable-forwarding` with 4.1 makes it clear what the default is, but I don't have a strong preference.

Reply:
Just breaking it down for sake of clarity:

* What default forwarding behavior does Zeek currently use ?  It does not use forwarding and we want to change that default after 4.0 so that it does (we're all in sync on the plans).

* What default forwarding behavior does Broker currently use ?  It *does* use forwarding (opposite of Zeek's current default config).  No changes are planned there.

* What is the name of the option that Broker exposes to alter forwarding behavior ?  There is currently `forward` (default of `true`), but that will be gone after merging this, so there's potential for user code that referred to that name/variable to become broken and that's an independent issue from the above points.  I was assuming (and still inferring from last response) that one can use the new `disable-forwarding` (default of `false) option for essentially the same goals as `forward`, so at least two ways we can handle this are (1) go back to using `forward` for the name or (2) call out the name difference in the release notes.  I'm planning to do (2) unless there's contrary feedback/corrections.

Reply:
Good summary. I'm fine with either option, btw. Whichever you prefer.